### PR TITLE
Update Licenses available for add-on developers to choose from on AMO

### PR DESCRIPTION
**Pull request by bakulf**
Fixes: mozilla/addons#15092

### Description

This PR introduces a new set of licenses and orders them all.

1. The new licenses can be imported via a new `import_new_licenses` command. This needs to be executed in prod.
2. a migration is used to sort the existing licenses

### Testing

No new tests were added, but several changes were included in the PR to make the existing tests pass.

### Checklist

- [x] Add `#ISSUENUM` at the top of your PR to an existing open issue in the mozilla/addons repository.
- [x] Successfully verified the change locally.
- [x] The change is covered by automated tests, or otherwise indicated why doing so is unnecessary/impossible.
- [x] Add before and after screenshots (Only for changes that impact the UI).
- [x] Add or update relevant [docs](../docs/) reflecting the changes made.

image

### CODE REVIEW

File: docs/topics/api/licenses.rst [hunk 1]
```diff
@@ -26,12 +26,16 @@ Open source license slugs are taken from the `SPDX License list <https://spdx.or
     ===================  ==============================================================
     all-rights-reserved  All Rights Reserved. This is not an open source license.
```
Comment (KevinMind) [inline docs/topics/api/licenses.rst:27]: Would it be possible to convert this document to .md. We have this issue with the stated goal to remove .rst and replace with .md files. A common approach to this is to convert a file when you touch it https://github.com/mozilla/addons/issues/9389

Not a requirement but could be an additional nice to have.

File: src/olympia/applications/management/commands/import_new_licenses.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+from django.core.management.base import BaseCommand
+
+from olympia.versions.models import License
+
+
+# This command is meant to be removed in 1 cycle after the importing/sorting of
+# a new set of licenses.
+
+
+class Command(BaseCommand):
```
[the remark this answers is not in the document]
Comment (bakulf) [reply src/olympia/applications/management/commands/import_new_licenses.py:11]: We must import the license to prod in order to show the new ones. We have to run this command to prod.

[the remark this answers is not in the document]
Comment (KevinMind) [reply src/olympia/applications/management/commands/import_new_licenses.py:11]: Your migration will get run in prod once this PR is shipped... so I'm not sure why we need a command for it.

[the remark this answers is not in the document]
Comment (bakulf) [reply src/olympia/applications/management/commands/import_new_licenses.py:11]: The migration does not create new licenses. It changes the built-in values of anything already existing in the DB, which is consistent with what we have for the previous licenses.

File: docs/topics/api/licenses.rst (hunk 1 above)
Comment (bakulf) [reply docs/topics/api/licenses.rst:27]: Let's do this in a separate commit.

**Comment by willdurand** [comment]
I also have 2 licenses that are the same (without this patch applied). Could it be a pre-existing bug?

Screenshot 2024-11-20 at 10 26 58

**Comment by KevinMind** [comment]
This is likely due to the existing [bug](https://github.com/mozilla/addons/issues/15153)

I was able to get what I believe is the correct state by "working around" that bug. Steps are:

> IMPORTANT: backup your data first if you care about it this will wipe everything

1. `make down`
2. `make docker_mysqld_volume_remove`
3. `make up`
4. Check the licenses

The current make up flow is broken in terms of producing reproducible data using INIT_CLEAN because it only flushes the database and does not totally wipe it. This causes migrations to not always run, which can change the db state. @bakulf can you confirm the screenshot here is the expected state and update the PR with the more accurate testing instructions.

image

**Comment by bakulf** [comment]
> I also have 2 licenses that are the same (without this patch applied). Could it be a pre-existing bug?


Correct. This bug is fixed in this PR. If you see the initial.json file contains builtin=1 twice.

**Comment by bakulf** [comment]
So yes, the list of licenses is described in https://github.com/mozilla/addons/issues/15092. I updated the screenshot in the PR description. In theory, you will see the new list of licenses by applying this patch and running a migration (or creating a fresh new environment).

Without this patch, the MPL-2.0 is duplicated because of a bug here: https://github.com/mozilla/addons-server/blob/master/src/olympia/versions/fixtures/initial.json#L76-L131 see the two `builtin=1`.

For better testing, I suggest the following:

*New env*
1. create a fresh new env: `make down; make docker_mysqld_volume_remove; make up`
2. set the `enable-submissions` flag
3. submit a new add-on.
4. you will see the new licenses in the correct order, without dups
5. no errors will be shown during the entire submission process.

*Existing env*
1. clone the repo without this PR
2. create a fresh new env: `make down; make docker_mysqld_volume_remove; make up`
2. set the `enable-submissions` flag
3. submit a new add-on
4. you will see the dup licenses bug
6. apply this PR
7. submit a new add-on
8. you will see the new licenses in the correct order, without dups
9. no errors will be shown during the entire submission process.

**Review by KevinMind — APPROVED** [review]
Looks good to me. I think we can revert the `import_license` command after https://github.com/mozilla/addons-server/pull/22860 lands but for now this should be fine.

**Pull request merged**